### ISSUE TITLE
v055c: Wireguard tunnel traffic (especially upload) incredibly slow since update

### ISSUE THREAD

**Issue by codiflow** [celzero/rethink-app#1306]
I recognized this behaviour on one of my devices so I started investigating if it has to do with the recent update. I used the second device where I use RethinkDNS too for the investigation. And I can say – yes, the update to v055c slows down any wireguard tunnel traffic to incredibly slow values.

On both devices I **upgraded from v055a to v055c** and the only thing I **changed** after the update was to open the Wireguard config (it was set to "Advanced" already) and activated "Lockdown" and "Always-on".

With the new version the speed of the tunnel went down to **incredibly slow values**. **Switching back** to the **old version v055a** brings back the "old speed":

Without Wireguard: 20.55 MBit/s (Download), 13.09 MBit/s (Upload), 13.2 (Jitter), 59 (Ping)
v055a: 2.0 MBit/s (Download), 8.3 MBit/s(Upload), 91.64 (Jitter), 169 (Ping)
v055c: 3.5 MBit/s (Download), 0.04 MBit/s (Upload), 43.9 (Jitter), 167 (Ping)

In used **the same WIFI** and the **same LibreSpeed server** to test everything to have reliable test results for you.

If you need more details on that issue just tell me 😎

**Comment by ignoramous** [comment]
Thanks.

We have since fixed packet drops (https://github.com/celzero/rethink-app/issues/1252), but unsure if that's what is spoiling things for you.

If you're technical enough, turn ON `Verbose` in *Configure* -> *Settings* -> *Log level*, then use `adb logcat | grep "wg:"` to see if something looks suspicious, and share it here if you're comfortable doing so?

**Comment by codiflow** [comment]
Thanks for your answer – looks like it's the same issue you mentioned above. I see some (not that many but they are there) `dropped pkt;` in the logs in verbose mode.

So #1252 will be fixed in the next version then?

Apart from that I want to say a big **THANK YOU** regarding the app in general. The recent Wireguard improvements are great and apart from the upload speed drop I had absolutely **no issues updating** 🥳.

**Comment by ignoramous** [comment]
> So #1252 will be fixed in the next version then?

Yes, due in a few days.

> see some (not that many but they are there) dropped pkt; in the logs in verbose mode.

Nothing else suspicious, correct?

**Comment by codiflow** [comment]
> Nothing else suspicious, correct?

Nope, everything fine and "green" in the (`wg:` filtered) logs so far 😎

**Comment by ignoramous** [comment]
Should have been fixed with [`v055d`](https://old.reddit.com/r/rethinkdns/comments/1btd7nn/v055d_shiny_new_wireguard/) (we are also planning on bringing in GRO/GSO related improvements from upstream, but it isn't immediate priority since in our tests, Rethink's WireGuard-as-a-proxy crosses 60mbps, which perhaps is *enough* for any handheld Android).

(feel free to re-open if that's not the case).

**Comment by codiflow** [comment]
Can confirm, recent update has fixed the issue. Thanks to the devs 🚀🙏
